contracts/native-precompiles/EcrecoverCaller.sol

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.5.0 <0.9.0;
+
+contract EcrecoverCaller {
+    event EcrecoverResult(bytes result);
+
+    function function1(bytes32 messageHash, uint8 v, bytes32 r, bytes32 s) external view returns (address) {
```

Comment:
Please give the functions `function1`, `function2` and `function3` names that relate to the result they are trying to achieve